File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int l = 0;
        int r = 0;
        int s = 0;
        int res = 0;
        while(r<arr.size()){
            // cout <<"s "<<s<<" r "<<r<<endl;
            while((r-l)<k){
                s+=arr[r];
                // cout <<"s "<<s<<" r "<<r<<endl;
                r+=1;
            }
            if(s>=k*threshold){
                // cout<<"entrd"<<endl;
                res+=1;
            }
            s-=arr[l];
            l+=1;
            // cout <<"s "<<s<<" l "<<l<<" r "<<r<<endl;
            // r+=1;
        }
        return res;
    }
};

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        l, r = 0, 0
        sm = 0
        res = 0
        while r < len(arr):
            if (r-l) < k:
                sm += arr[r]
                r += 1
                if sm/k >= threshold and (r - l) == k:
                    # print("l", l,arr[l],"r", r,arr[r],"s", sm,"a", sm/k, sm/k >= threshold)
                    res += 1
                continue
            sm -= arr[l]
            l += 1
            
        return res